### ISSUE TITLE
Integration into sverchok

### ISSUE THREAD

**Issue by Sinasta** [barckley75/Blender_Neural_Network#1]
First of all thank you very much for your work. i would like to ask you if it is possible to write this addon as a sverchok addon (https://github.com/nortikin/sverchok) ? In sverchok it is already possible with nodes to read many different text files (csv, json, txt) and it gives options to manipulate the importet CSV data. Furthermore there is a good documentation onto how to write addons for sverchok. My next question would be do you plan on integration Scikit-learn?

**Comment by Sinasta** [comment]
If you are interested in writing the addon for sverchok i would highly suggest you to open an issue on the sverchok github. The developers are very active and helpful.

**Comment by barckley75** [comment]
- I think, if we are talking about dataset, nodes are not the right way to go. My idea is to write everything in python without using any nodes environment. I'm preparing a road map for this topic.

- You can import any library in Blender, take a look here: https://stackoverflow.com/questions/65076829/how-to-use-tensorflow-in-blender/68335409#68335409

**Comment by Sinasta** [comment]
I think i missunderstood the purpose of your addon. If your purpose is to write everything in python without any nodes than i don't really understand why this is an addon for blender? Just for the visualisation of the Model in Blender? I was looking for an addon for blender to perform training with nodes. I thought your addon could eventually fulfill this. but if i missunderstood than im sorry for opening an issue and i will close it.

**Comment by Sinasta** [comment]
I re-read the readme and i think i understood. You want only the sidepanel not any nodes. I get that choice. i still think an integration into sverchok would give some flexebility but i get your choice.

**Comment by barckley75** [comment]
I don't think one choice could exclude the other. But what I understood for training purposes it is easier to interact with a UI such as a panel. I love nodes and I'm taking in consideration to write some custom nodes specifically for splitting datasets for training, validation and test. But it is really complicate to understand for other users what's going on in the node tree when it growths, and trust me, a node tree could growth very fast, and you have to take in consideration that it's full of exceptions that it is not really convenient to face them with a nodes environment.
Said that, the purpose of my project is to give a UI for training models with different frameworks and at the same time have the possibility to create professional render animation of AI [models](https://www.youtube.com/watch?v=Aop4rGjMskI).
So, it is opened to nodes environment but I think they are more appropriate for the artistic part of the project.
What I really like is to take a look inside the neural network, usually we look at the cost, error and its behavior, in general, the parameters that track the efficiency, but I think assign, for example, the behavior of the weights e.g. to the height meshes we could have some interesting aesthetic surprise.
So, I'm dividing the training section, that basically is very structured, from the artistic representation part, that for sure is a good match with nodes, animation or sverchok.

**Comment by Sinasta** [comment]
You are right and im reallg exited to test your project! My only thought for the advantage of integrating the training part into sverchok would have been that the whole data manipulation part on the CSV files could be done in sverchok without using pandas etc. And another big advantage for me personally would have been that i could create a project with sverchok ( in my case an architectural project ) and than without leaving blender or sverchok directly train the model in the same node tree with the output data. But when i think more about it than this would still be possible with a sidepanel. Right now your addon is relying on tensorflow if i understand correctly. Do you have plans to integrate PyFlow or Scikit-learn?

**Comment by barckley75** [comment]
If you want to try yourself, to import Scikit-learn in Blender is very easy as mentions above. Once you import it, you can create a Script node in animation nodes, press "Shift +a > Subprograms > Script", create a file text in Blender Text area, and here you can read the variables from the tree, elaborate them and output the data processed to the tree.

**Comment by Sinasta** [comment]
Thanks for the information.
